src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -148,11 +160,91 @@ protected override void OnStopped()
             // the other is external thing disconnecting remote host from us (ex, user killing OOP process).
             // the Disconnected event we subscribe is to detect #2 case. and this method is for #1 case. so when we are willingly disconnecting
             // we don't need the event, otherwise, Disconnected event will be called twice.
+            UnregisterGlobalOperationNotifications();
             _rpc.Disconnected -= OnRpcDisconnected;
             _rpc.Dispose();
             _remotableDataRpc.Dispose();
         }
 
+        private void RegisterGlobalOperationNotifications()
+        {
+            var globalOperationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            if (globalOperationService != null)
+            {
+                globalOperationService.Started += OnGlobalOperationStarted;
+                globalOperationService.Stopped += OnGlobalOperationStopped;
+            }
+        }
+
+        private void UnregisterGlobalOperationNotifications()
+        {
+            var globalOperationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            if (globalOperationService != null)
+            {
+                globalOperationService.Started -= OnGlobalOperationStarted;
+                globalOperationService.Stopped -= OnGlobalOperationStopped;
+            }
+
+            lock (_globalNotificationsGate)
+            {
+                // Unilaterally transition us to the finished state.  Once we're finished
+                // we cannot start or stop anymore.
+                _globalNotificationsTask = _globalNotificationsTask.ContinueWith(
+                    _ => GlobalNotificationState.Finished, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
+            }
```

Comment:
this might have a race. I think this should wait for _globalNotificationTask to return.